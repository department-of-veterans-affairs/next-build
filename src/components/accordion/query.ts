import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAccordion } from '@/types/drupal/paragraph'
import { AccordionItem } from '@/components/accordion/formatted-type'
import { getHtmlFromField } from '@/lib/utils/getHtmlFromField'

export const formatter: QueryFormatter<ParagraphAccordion, AccordionItem> = (
  entity: ParagraphAccordion
) => {
  return {
    type: entity.type as AccordionItem['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    header: entity.field_header || null,
    html: getHtmlFromField(entity.field_rich_wysiwyg),
  }
}
