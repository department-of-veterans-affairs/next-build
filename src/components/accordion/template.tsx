import {
  Accordion as FormattedAccordion,
  AccordionItem as FormattedAccordionItem,
} from '@/components/accordion/formatted-type'
import { slugifyString } from '@/lib/utils/slug'
import { ParagraphComponent } from '@/components/paragraph/formatted-type'

export function AccordionItem({
  id,
  header,
  html,
  itemLevel,
}: ParagraphComponent<FormattedAccordionItem>) {
  return (
    <va-accordion-item
      key={`${id}-${slugifyString(header)}`}
      header={header}
      class="va-accordion-item"
      id={`${slugifyString(header)}`}
      data-testid={`accordion-item-${id}-${slugifyString(header)}`}
      level={itemLevel ?? 2}
    >
      {html && <div dangerouslySetInnerHTML={{ __html: html }} />}
    </va-accordion-item>
  )
}

export function Accordion({ id, bordered, items }: FormattedAccordion) {
  if (!items) return null
  return (
    <div>
      <va-accordion bordered={bordered} id={id}>
        {items.map((item) => (
          <AccordionItem key={item.id} {...item} />
        ))}
      </va-accordion>
    </div>
  )
}
