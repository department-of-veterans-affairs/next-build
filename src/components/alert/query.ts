import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAlert } from '@/types/drupal/paragraph'
import { Alert, AlertBlock, AlertType } from '@/components/alert/formatted-type'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { queries } from '@/lib/drupal/queries'

export const formatter: QueryFormatter<ParagraphAlert, Alert> = (
  entity: ParagraphAlert
) => {
  return {
    type: entity.type as Alert['type'],
    id: entity.id ?? null,
    entityId: entity.drupal_internal__id ?? null,
    alertType: entity.field_alert_type as AlertType,
    heading: entity.field_alert_heading,
    blockReference: queries.formatData(
      'block--alert',
      entity.field_alert_block_reference
    ) as AlertBlock,
    paragraphs: entity.field_va_paragraphs.map((paragraph) =>
      formatParagraph(paragraph)
    ),
  }
}
