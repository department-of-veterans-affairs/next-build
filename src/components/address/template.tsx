import { GoogleMapsDirections } from '@/components/googleMapsDirections/template'
import { FieldAddress } from '@/types/drupal/field_type'

interface AddressProps {
  address: FieldAddress
  title: string
  showOrganization?: boolean
}

export const Address = ({
  address,
  title,
  showOrganization = false,
}: AddressProps) => {
  return (
    <>
      <address>
        <p className="vads-u-margin-bottom--0 vads-u-margin-top--0">
          {showOrganization && address.organization && (
            <>
              {address.organization}
              <br />
            </>
          )}
          {address.address_line1}
          {address.address_line2 && (
            <>
              <br />
              {address.address_line2}
            </>
          )}
          <br />
          {`${address.locality}, ${address.administrative_area} ${address.postal_code}`}
        </p>
      </address>
      <p className="vads-u-margin-bottom--0 vads-u-margin-top--0">
        <GoogleMapsDirections address={address} location={title} />
      </p>
    </>
  )
}
