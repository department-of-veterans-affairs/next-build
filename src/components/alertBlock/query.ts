// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter, QueryParams } from 'next-drupal-query'
import { AlertBlock, AlertType } from '@/components/alert/formatted-type'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { Wysiwyg } from '@/components/wysiwyg/formatted-type'
import { ExpandableText } from '@/components/expandableText/formatted-type'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

export const params: QueryParams<null> = () =>
  new DrupalJsonApiParams().addInclude(['field_alert_content'])

export const formatter: QueryFormatter<BlockAlert, AlertBlock> = (
  entity: BlockAlert
) => {
  if (!entity) {
    return null
  }

  return {
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    id: entity.id,
    title: entity.field_alert_title,
    content: formatParagraph(entity.field_alert_content) as
      | Wysiwyg
      | ExpandableText,
    // {
    //   header:
    //     entity.field_alert_content.field_text_expander ??
    //     entity.field_alert_heading,
    //   text: entity.field_alert_content?.field_wysiwyg?.processed ?? '',
    // },
  }
}
