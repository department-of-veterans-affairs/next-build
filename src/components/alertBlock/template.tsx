import {
  VaAlert,
  VaAlertExpandable,
} from '@department-of-veterans-affairs/component-library/dist/react-bindings'
import { AlertBlock as FormattedAlertBlock } from '@/types/formatted/alert'
import { PARAGRAPH_RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { Wysiwyg } from '@/components/wysiwyg/template'

export function AlertBlock(alertBlock: FormattedAlertBlock) {
  if (!alertBlock) {
    return null
  }

  const { alertType, id, title, content } = alertBlock
  return (
    <>
      <VaAlert id={id} status={alertType} class="vads-u-margin-top--3" uswds>
        <h2 slot="headline" className="vads-u-font-size--h3">
          {title}
        </h2>

        {content.type === PARAGRAPH_RESOURCE_TYPES.WYSIWYG && (
          <Wysiwyg {...content} />
        )}

        {content.type === PARAGRAPH_RESOURCE_TYPES.EXPANDABLE_TEXT && (
          <VaAlertExpandable id={id} trigger={content.header}>
            {content.text && (
              <div
                dangerouslySetInnerHTML={{
                  __html: content.text,
                }}
              />
            )}
          </VaAlertExpandable>
        )}
      </VaAlert>
    </>
  )
}
