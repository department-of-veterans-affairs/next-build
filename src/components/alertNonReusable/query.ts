// Define the query params for fetching block--alert.
import { QueryFormatter } from 'next-drupal-query'
import { AlertType, AlertNonReusable } from '@/components/alert/formatted-type'
import { ParagraphNonReusableAlert } from '@/types/drupal/paragraph'
import { formatParagraph } from '@/lib/drupal/paragraphs'

export const formatter: QueryFormatter<
  ParagraphNonReusableAlert,
  AlertNonReusable
> = (entity: ParagraphNonReusableAlert) => {
  if (!entity) {
    return null
  }

  return {
    type: entity.type as AlertNonReusable['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    heading: entity.field_alert_heading,
    paragraphs: entity.field_va_paragraphs.map((paragraph) => formatParagraph(paragraph)),
  }
}
