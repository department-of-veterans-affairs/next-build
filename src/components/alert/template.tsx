import React from 'react'
import { Alert as FormattedAlert } from '@/components/alert/formatted-type'
import { ParagraphComponent } from '@/components/paragraph/formatted-type'
import { AlertBlock } from '@/components/alertBlock/template'
import { Paragraph } from '@/components/paragraph/template'

export function Alert(alert: ParagraphComponent<FormattedAlert>) {
  if (!alert) {
    return null
  }

  const { entityId, alertType, heading, blockReference, paragraphs } = alert

  if (blockReference) {
    return <AlertBlock {...blockReference} />
  }

  return (
    <va-alert
      data-template="paragraphs/alert"
      data-paragraph-type="paragraph--alert"
      data-entity-id={entityId}
      status={alertType}
      class="vads-u-margin-top--3"
      uswds
    >
      <h2 slot="headline" className="vads-u-font-size--h3">
        {heading}
      </h2>

      {paragraphs?.map((paragraph) => (
        <Paragraph key={paragraph.id || paragraph.entityId} {...paragraph} />
      ))}
    </va-alert>
  )
}

export default Alert
