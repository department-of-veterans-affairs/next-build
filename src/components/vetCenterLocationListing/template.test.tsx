import React from 'react'
import { render, screen } from '@testing-library/react'
import { VetCenterLocationListing } from './template'
import { formatter } from './query'
import drupalMockData from './mock.json'
import mockCap from './mock.vetCenterCap.json'
import mockOutstation from '@/components/vetCenterOutstation/mock.json'
import { NodeVetCenterCap } from '@/types/drupal/node'
import { NodeVetCenterOutstation } from '@/types/drupal/node'

// Restructure mock data to match expected format
const mockData = formatter({
  entity: drupalMockData,
  // @ts-expect-error drupalMockData technically has numbers instead of strings
  // for some of the IDs, but this is a known problem. See
  // https://github.com/chapter-three/next-drupal/issues/686#issuecomment-2083175598
  caps: [mockCap as NodeVetCenterCap],
  // @ts-expect-error drupalMockData technically has numbers instead of strings
  // for some of the IDs, but this is a known problem. See
  // https://github.com/chapter-three/next-drupal/issues/686#issuecomment-2083175598
  outstations: [mockOutstation as NodeVetCenterOutstation],
  mobileVetCenters: [],
})

// Mock window object
Object.defineProperty(window, 'mainVetCenterPhone', {
  value: '',
  writable: true,
})
Object.defineProperty(window, 'mainVetCenterAddress', {
  value: {},
  writable: true,
})
Object.defineProperty(window, 'mainVetCenterId', {
  value: '',
  writable: true,
})
Object.defineProperty(window, 'satteliteVetCenters', {
  value: [],
  writable: true,
})

describe('VetCenterLocationListing with valid data', () => {
  test('renders VetCenterLocationListing component', () => {
    render(<VetCenterLocationListing {...mockData} />)

    expect(screen.queryByText(/Locations/)).toBeInTheDocument()
  })

  describe('Section headings', () => {
    test('renders main location heading', () => {
      render(<VetCenterLocationListing {...mockData} />)

      expect(
        screen.getByRole('heading', { level: 2, name: /main location/i })
      ).toBeInTheDocument()
    })

    test('renders satellite locations heading when satellite locations exist', () => {
      render(<VetCenterLocationListing {...mockData} />)

      expect(
        screen.getByRole('heading', { level: 2, name: /satellite locations/i })
      ).toBeInTheDocument()
    })

    test('renders vet centers in other areas heading', () => {
      render(<VetCenterLocationListing {...mockData} />)

      expect(
        screen.getByRole('heading', {
          level: 2,
          name: /vet centers in other areas/i,
        })
      ).toBeInTheDocument()
    })

    test('does not render satellite locations heading when no satellite locations exist', () => {
      const mockDataNoSatellites = {
        ...mockData,
        satelliteLocations: [],
        nearbyMobileVetCenters: [],
        mobileVetCenters: [],
      }
      render(<VetCenterLocationListing {...mockDataNoSatellites} />)

      expect(
        screen.queryByRole('heading', {
          level: 2,
          name: /satellite locations/i,
        })
      ).not.toBeInTheDocument()
    })
  })

  describe('Main office display', () => {
    test('renders main office information', () => {
      render(<VetCenterLocationListing {...mockData} />)

      // Check for main office title within va-link using data attributes
      const titleLinkElement = document.querySelector(
        'va-link[text="Colorado Springs Vet Center"]'
      )
      expect(titleLinkElement).toBeInTheDocument()
      expect(titleLinkElement).toHaveAttribute(
        'href',
        '/colorado-springs-vet-center'
      )

      // Check for address information
      expect(screen.getByText(/3920 North Union Blvd/)).toBeInTheDocument()
      expect(screen.getByText(/Colorado Springs, CO 80907/)).toBeInTheDocument()
    })

    test('renders main office phone number', () => {
      render(<VetCenterLocationListing {...mockData} />)

      // Phone number is rendered as va-telephone element
      const phoneElement = document.querySelector(
        'va-telephone[contact="7194719992"]'
      )
      expect(phoneElement).toBeInTheDocument()
    })
  })

  describe('Satellite vet center types', () => {
    test('renders CAP (Community Access Point) location', () => {
      render(<VetCenterLocationListing {...mockData} />)

      // Check for CAP location title
      expect(
        screen.getByText(/Little Rock Vet Center - Mountain Home/)
      ).toBeInTheDocument()

      // Check for CAP address with organization (when organization exists)
      expect(
        screen.getByText(/St. Peter the Fisherman Church/)
      ).toBeInTheDocument()
      expect(screen.getByText(/249 Dyer St/)).toBeInTheDocument()

      // Check for CAP-specific operating status info
      expect(
        screen.getByText(/Counseling services, including a PTSD support group/)
      ).toBeInTheDocument()
    })

    test('renders outstation location', () => {
      render(<VetCenterLocationListing {...mockData} />)

      // Check for outstation title
      expect(
        screen.getByText(/Sepulveda Vet Center Outstation/)
      ).toBeInTheDocument()

      // Check for outstation address
      expect(screen.getByText(/9737 Haskell Ave/)).toBeInTheDocument()
    })

    test('renders nearby mobile vet center', () => {
      render(<VetCenterLocationListing {...mockData} />)

      // Check for nearby mobile vet center title
      expect(screen.getByText(/Pueblo Mobile Vet Center/)).toBeInTheDocument()

      // Mobile vet centers should not display addresses
      expect(
        screen.queryByText(/1515 Fortino Boulevard/)
      ).not.toBeInTheDocument()
    })
  })

  describe('Find locations link', () => {
    test('renders find vet center location link', () => {
      render(<VetCenterLocationListing {...mockData} />)

      // Look for the va-link element with the correct text and href using querySelector
      const linkElement = document.querySelector(
        'va-link[text="Find a Vet Center location"]'
      )
      expect(linkElement).toBeInTheDocument()
      expect(linkElement).toHaveAttribute('href', '/find-locations')
    })
  })

  describe('Global variables for nearby vet centers widget', () => {
    beforeEach(() => {
      // Reset window properties before each test to undefined (more accurate initial state)
      window.mainVetCenterPhone = undefined
      window.mainVetCenterAddress = undefined
      window.mainVetCenterId = undefined
      window.satteliteVetCenters = undefined
    })

    test('sets up window variables with the provided props', () => {
      render(<VetCenterLocationListing {...mockData} />)

      expect(window.mainVetCenterPhone).toBe(mockData.mainOffice.phoneNumber)
      expect(window.mainVetCenterAddress).toMatchSnapshot()
      expect(window.mainVetCenterId).toBe(
        mockData.mainOffice.fieldFacilityLocatorApiId
      )
      expect(window.satteliteVetCenters).toEqual(
        mockData.satelliteLocations.map(
          (location) => location.fieldFacilityLocatorApiId
        )
      )
    })
  })
})
