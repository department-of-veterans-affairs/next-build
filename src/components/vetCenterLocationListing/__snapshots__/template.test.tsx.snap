// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`VetCenterLocationListing with valid data Global variables for nearby vet centers widget sets up window variables with the provided props 1`] = `
{
  "addressLine1": "3920 North Union Blvd",
  "addressLine2": "Suite 310",
  "administrativeArea": "CO",
  "locality": "Colorado Springs",
  "postalCode": "80907",
}
`;
